packages/flutter_tools/lib/src/project.dart

```diff
@@ -73,84 +82,59 @@ class FlutterProject {
-  IosModuleProject get iosModule => new IosModuleProject(directory.childDirectory('.ios'));
-
-  File get androidLocalPropertiesFile {
-    return directory
-        .childDirectory(manifest.isModule ? '.android' : 'android')
-        .childFile('local.properties');
-  }
-
-  File get generatedXcodePropertiesFile {
-    return directory
-        .childDirectory(manifest.isModule ? '.ios' : 'ios')
-        .childDirectory('Flutter')
-        .childFile('Generated.xcconfig');
-  }
+  AndroidProject get android => new AndroidProject._(_androidDirectory, isModule);
 
   File get flutterPluginsFile => directory.childFile('.flutter-plugins');
 
-  Directory get androidPluginRegistrantHost {
-    return manifest.isModule
-        ? directory.childDirectory('.android').childDirectory('Flutter')
-        : directory.childDirectory('android').childDirectory('app');
-  }
-
-  Directory get iosPluginRegistrantHost {
-    // In a module create the GeneratedPluginRegistrant as a pod to be included
-    // from a hosting app.
-    // For a non-module create the GeneratedPluginRegistrant as source files
-    // directly in the iOS project.
-    return manifest.isModule
-        ? directory.childDirectory('.ios').childDirectory('Flutter').childDirectory('FlutterPluginRegistrant')
-        : directory.childDirectory('ios').childDirectory('Runner');
-  }
-
   /// The example sub-project of this project.
-  FlutterProject get example => new FlutterProject._(_exampleDirectory, _exampleManifest, FlutterManifest.empty());
+  FlutterProject get example => new FlutterProject(
+    _exampleDirectory(directory),
+    _exampleManifest,
+    FlutterManifest.empty(),
+  );
+
+  bool get isModule => manifest != null && manifest.isModule;
 
-  /// True, if this project has an example application
-  bool get hasExampleApp => _exampleDirectory.childFile('pubspec.yaml').existsSync();
+  /// True, if this project has an example application.
+  bool get hasExampleApp => _exampleDirectory(directory).existsSync();
 
   /// The directory that will contain the example if an example exists.
-  Directory get _exampleDirectory => directory.childDirectory('example');
+  static Directory _exampleDirectory(Directory directory) => directory.childDirectory('example');
+
+  /// The directory that contains the Android app to be launched by `flutter run`.
+  Directory get _androidDirectory => directory.childDirectory(isModule ? '.android' : 'android');
+
+  /// The directory that contains the iOS app to be launched by `flutter run`.
+  Directory get _iosDirectory => directory.childDirectory(isModule ? '.ios' : 'ios');
 
   /// Generates project files necessary to make Gradle builds work on Android
   /// and CocoaPods+Xcode work on iOS, for app and module projects only.
   Future<void> ensureReadyForPlatformSpecificTooling() async {
-    if (!directory.existsSync() || hasExampleApp) {
+    if (!directory.existsSync() || hasExampleApp)
       return;
```

Comment:
Move return to previous line?

Reply:
I believe I'm following the style guide here: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#separate-the-if-expression-from-its-statement